app/controllers/api/cloud_networks_controller.rb

```diff
@@ -9,5 +9,31 @@ def delete_resource(type, resource_id, _data = {})
       task_id = cloud_network.delete_cloud_network_queue(User.current_user.userid)
       action_result(true, "Deleting #{cloud_network.name}", :task_id => task_id)
     end
+
+    def options_by_ems_id
```

Comment:
Can you make `options_by_ems_id` and `options_by_id` private? (Either prefix with `private` or put them in a `private` section at the bottom).